### PR TITLE
ci: Prepare release 3.1.0

### PR DESCRIPTION
**Pull request by fsbraun**
## Description

This PR bumps the version and updates the changelog.

## What's Changed
 * Add GitHub Actions by @crydotsnake in https://github.com/django-cms/djangocms-style/pull/56
 * FEATURE: adjust setup metadata by @crydotsnake in https://github.com/django-cms/djangocms-style/pull/58
 * FEATURE: add github pr template by @crydotsnake in https://github.com/django-cms/djangocms-style/pull/57
 * Removed pinned django-treebeard by @tidenhub in https://github.com/django-cms/djangocms-style/pull/59
 * Feat: Django 4.2 support by @mogoh in https://github.com/django-cms/djangocms-style/pull/61
 * chore: Improve ruff config by @marksweb in https://github.com/django-cms/djangocms-style/pull/64

 ## New Contributors
 * @crydotsnake made their first contribution in https://github.com/django-cms/djangocms-style/pull/56
 * @tidenhub made their first contribution in https://github.com/django-cms/djangocms-style/pull/59
 * @mogoh made their first contribution in https://github.com/django-cms/djangocms-style/pull/61
 * @marksweb made their first contribution in https://github.com/django-cms/djangocms-style/pull/64

## Checklist

* [x] I have opened this pull request against ``master``
* [ ] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on
[Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

**Comment by fsbraun** [comment]
@marksweb I've updated the flows accordingly. Essentially, the login info can go.

**Comment by marksweb** [comment]
> @marksweb I've updated the flows accordingly. Essentially, the login info can go.

It's a bit more than that.

File: .github/workflows/publish-to-live-pypi.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    types:
+      - published
+
+jobs:
+  build-n-publish:
+    name: Build and publish Python 🐍 distributions 📦 to pypi
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/djangocms-snippet
```
Comment (marksweb) [inline .github/workflows/publish-to-live-pypi.yml:14]: This needs to match what the pypi project is (or will be when creating these for new projects)

**Review by marksweb — CHANGES_REQUESTED** [review]

**Comment by fsbraun** [comment]
Right, snippet != style. Sorry.

**Review by marksweb — APPROVED** [review]

**Pull request merged**